crates/sui-indexer/src/db.rs

```diff
@@ -263,55 +278,96 @@ mod tests {
     #[tokio::test]
     async fn db_migration_consistency_smoke_test() {
         let database = TempDb::new().unwrap();
-        let blocking_pool = new_connection_pool(
-            database.database().url().as_str(),
-            &ConnectionPoolConfig::default(),
+        let pool = ConnectionPool::new(
+            database.database().url().to_owned(),
+            ConnectionPoolConfig {
+                pool_size: 2,
+                ..Default::default()
+            },
         )
+        .await
         .unwrap();
-        let mut conn = get_pool_connection(&blocking_pool).unwrap();
-        reset_database(&mut conn).await.unwrap();
-        check_db_migration_consistency(&mut conn).unwrap();
+
+        reset_database(pool.dedicated_connection().await.unwrap())
+            .await
+            .unwrap();
+        check_db_migration_consistency(&mut pool.get().await.unwrap())
```

Comment:
Is there a reason one uses dedicated while the other use non-dedicated?

Reply:
yes, reset_database requires an owned connection that is passed by value while the consistency check just needs one by &mut